packages/wallets/src/components/FlowProvider/FlowProvider.tsx

```diff
@@ -66,39 +66,32 @@ export const useFlow = () => {
  */
 function FlowProvider<T extends TWalletScreens>({
     children,
+    initialScreenId,
     initialValues,
     screens,
-    screensOrder,
 }: TFlowProviderProps<T>) {
-    const [currentScreenId, setCurrentScreenId] = useState<keyof T>(screensOrder[0]);
+    const [currentScreenId, setCurrentScreenId] = useState<keyof T>(initialScreenId || Object.keys(screens)[0]);
     const switchScreen = (screenId: keyof T) => {
         setCurrentScreenId(screenId);
     };
 
     const FlowProvider = FlowProviderContext.Provider as React.Provider<TFlowProviderContext<T> | null>;
-    const currentScreenIndex = useMemo(() => screensOrder.indexOf(currentScreenId), [currentScreenId, screensOrder]);
-    const isFinalScreen = currentScreenIndex >= screensOrder.length - 1;
-
-    const switchNextScreen = () => {
-        if (!isFinalScreen) {
-            const nextScreenId = screensOrder[currentScreenIndex + 1];
-            switchScreen(nextScreenId);
-        }
-    };
 
     const currentScreen = useMemo(() => {
         return screens[currentScreenId];
     }, [currentScreenId, screens]);
 
-    if (!currentScreenId) return null;
-
     const context = {
         currentScreenId,
-        switchNextScreen,
         switchScreen,
         WalletScreen: currentScreen,
     };
 
+    useEffect(() => {
+        if (initialScreenId) setCurrentScreenId(initialScreenId);
+    }, [initialScreenId]);
+
+    if (!currentScreenId) return null;
     return (
         // We let the logic of the onSubmit be handled by the flow component
         <Formik initialValues={initialValues} onSubmit={() => undefined}>
```

Comment:
not related to the current changes tho I've a query about this why the submit will be handled by the flow component?
If formik is being used, then the submit handle should be used to trigger validation as well unless we want to trigger validations manually, then using formik in this context is just an extra step which doesn't do anything

Reply:
If this will be a step form, then every component can be different form handling it's own initial values and validations then move to the next step?

Reply:
Also `setFieldValue, values` are being passed as props, rather `useFormikContext` can be used any of the child component of Formik to get all of the values and actions, if this approach will be used

Reply:
We'll soon add validation to the flow provider, but the submission is handled in the parent component (i.e. the component returning the FlowProvider) so the logic for all the flow navigation is visible in one component.